front/lib/api/assistant/agent.ts

```diff
@@ -216,6 +219,95 @@ export async function* runAgent(
   | AgentGenerationSuccessEvent
   | AgentMessageSuccessEvent
 > {
+  // First run the action if a configuration is present.
+  if (configuration.action !== null) {
+    if (isRetrievalConfiguration(configuration.action)) {
+      const eventStream = runRetrieval(
+        auth,
+        configuration,
+        conversation,
+        userMessage,
+        agentMessage
+      );
+
+      for await (const event of eventStream) {
+        if (event.type === "retrieval_params") {
+          yield event;
+        }
+        if (event.type === "retrieval_documents") {
+          yield event;
+        }
+        if (event.type === "retrieval_error") {
+          yield {
+            type: "agent_error",
+            created: event.created,
+            configurationId: configuration.sId,
+            messageId: agentMessage.sId,
+            error: {
+              code: event.error.code,
+              message: event.error.message,
+            },
+          };
+        }
+        if (event.type === "retrieval_success") {
+          yield {
+            type: "agent_action_success",
+            created: event.created,
+            configurationId: configuration.sId,
+            messageId: agentMessage.sId,
+            action: event.action,
+          };
+
+          // We stitch the action into the agent message. The conversation is expected to include
+          // the agentMessage object, updating this object will update the conversation as well.
+          agentMessage.action = event.action;
+        }
+      }
+    } else {
+      throw new Error(
```

Comment:
Nit: would be nicer to throw early instead of indenting the whole code inside the if (easier to read / understand if I can pop that branch off my mental stack)

Reply:
In the future with more actions this construct will make more sense?